### PR TITLE
K8 Tweaks!

### PR DESCRIPTION
**Pull request by gabrielsaruhashi**
## Description
- Kubernetes Integration
- Naive first attempt at supporting multiple visitation types
- E-Messaging scaffolding

### CODE REVIEW

File: Dockerfile.prod [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+# Copy remainder of the source code into the container
+COPY --chown=docker:docker . /app/src/
+
+RUN npm run build
+
+COPY --chown=docker:docker .env .env
+
+EXPOSE 5000
+CMD npm run start:dev
```
Comment (lspdrz) [inline Dockerfile.prod:70]: why start:dev and not just start?

File: docker-compose.yml [hunk 2]
```diff
@@ -1,32 +1,29 @@
...
-  #     - .env
-  #   volumes:
-  #     - 'mysqldata:/var/lib/mysql' # Use a docker volume for the database files
-
+  redis:
+    image: redis:alpine
+    ports:
+      - 6379:6379
+  mysql:
+    image: mysql:latest # Should we commit to a particular version?
```
Comment (lspdrz) [inline docker-compose.yml:8]: should we commit to a mysql version before shipping? probably

File: docker-compose.yml [hunk 3]
```diff
@@ -50,7 +46,39 @@ services:
...
+      MYSQL_PORT: '3307'
+      MYSQL_DATABASE: 'connect'
+      MYSQL_USER: 'admin'
+      MYSQL_PASSWORD: 'password'
+      MYSQL_ROOT_PASSWORD: 'password'
+      MYSQL_HOST: 'mysql'
+      JWT_SECRET: 'jwt_secret'
+      JWT_EXPIRATION_TIME: '1200'
+      SESSION_SECRET: '1234'
+      LETTERS_API_HOST: 'https://letters-api-staging.ameelio.org/api/'
```
Comment (lspdrz) [inline docker-compose.yml:66]: should this stuff be public in the repo?

File: src/calls/entities/call.entity.ts [hunk 4]
```diff
@@ -17,9 +20,14 @@ import VideoHandler from '@src/videoHandler/entities/videoHandler.entity';
 import Inmate from '@src/inmates/entities/inmate.entity';
 import User from '@src/users/entities/user.entity';
 import { SchedulerType } from '../enums/schedulerType.enum';
+import { VisitationType } from '../enums/visitationType.enum';
 
 @Entity({ name: 'Call' })
 class Call extends BaseEntity {
+  // If we migrate to Postgres, we can just do Generate('increment') as Postgres allows multiple primary keys
+  @Column({ unique: true })
+  index: number;
```
Comment (lspdrz) [inline src/calls/entities/call.entity.ts:29]: what does this index track?

File: src/users/users.service.ts [hunk 5]
```diff
@@ -37,6 +39,7 @@ export default class UsersService extends CrudService<User> {
     private readonly filesService: FilesService,
     private readonly devicesService: DevicesService,
     private readonly notificationsService: NotificationsService,
+    private readonly messengerService: MessengerServices,
```
Comment (lspdrz) [inline src/users/users.service.ts:42]: nit: MessengerService?

**Review by lspdrz — COMMENTED** [review]
Reviewed @gabrielsaruhashi !!!

File: docker-compose.yml (hunk 2 above)
Comment (gabrielsaruhashi) [reply docker-compose.yml:8]: yep - we committed to 8.0 for the pod
I'll update docker compose

File: Dockerfile.prod (hunk 1 above)
Comment (gabrielsaruhashi) [reply Dockerfile.prod:70]: oh was just copy pasting. ya that woks

File: src/calls/entities/call.entity.ts (hunk 4 above)
Comment (gabrielsaruhashi) [reply src/calls/entities/call.entity.ts:29]: https://ameelioorg.slack.com/archives/C023NSPKWV9/p1622852311295800

**Pull request merged**